src/net/gateway-discovery.ts

```diff
@@ -116,6 +120,11 @@ export class GatewayDiscovery {
                 // Sort nodes
                 const sortedNodes = this.sortNodes(healthyNodes)
 
+                // Set the archive IPNS Id if given
+                if (this.archiveIpnsId) {
+                    sortedNodes.web3.map((web3Gateway: Web3Gateway) => web3Gateway.archiveIpnsId = this.archiveIpnsId)
```

Comment:
This will do the thing, but the intended way is using `.forEach()`

`map` uses the return value and creates an output array (which we don't need/use)